### PR TITLE
Week 12a - UI Improvements

### PR DESCRIPTION
**Pull request by connienguyen**
+ Removed functionality for anonymous user event registrations
+ Updated the UI on event registration to match `User` properties
+ Add functionality to switch themes in Settings
+ Updated UI on storyboards so that UI is proportional on different devices
+ Add missing assets for tab bar navigation

### CODE REVIEW

File: VOLA/VOLA/Views/Cells/ThemeManagementCell.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        for theme in Theme.availableThemes {
+            segmentedThemeControl.setTitle(theme.themeName, forSegmentAt: theme.rawValue)
+        }
+        segmentedThemeControl.selectedSegmentIndex = ThemeManager.shared.currentTheme.rawValue
+    }
+}
+
+// MARK: - IBActions
+extension ThemeManagementCell {
+    @IBAction func onSegmentedControlSelect(sender: AnyObject) {
```
Comment (bphenriques) [inline VOLA/VOLA/Views/Cells/ThemeManagementCell.swift]: You should delegate this type of behaviour (trigger theme change if and only if the theme has indeed changed) to the theme manager.

Tip. You can use the `didSet` in conjunction with `guard value != oldValue`

File: VOLA/VOLA/Views/Cells/ThemeManagementCell.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        }
+
+        // Future improvement: apply theme as a preview and save changes
+        // to currentTheme after user approves preview
+        ThemeManager.shared.apply(theme)
+        ThemeManager.shared.saveTheme(theme)
+
+        // Post notification so UI elemented not affected by UIAppearance proxy changes are
+        // updated to match theme
+        NotificationCenter.default.post(name: NotificationName.themeDidChange, object: nil)
```
Comment (bphenriques) [inline VOLA/VOLA/Views/Cells/ThemeManagementCell.swift]: The notification should be sent by the manager and not by the UI

File: VOLA/VOLA/Helpers/Util/ThemeManager.swift [hunk 3]
```diff
@@ -51,14 +51,24 @@ final class ThemeManager {
...
         VolunteersNeededLabel.appearance().backgroundColor = theme.accentColor
         VolunteersNeededLabel.appearance().textColor = theme.tintColor
         RegisteredLabel.appearance().textColor = theme.tintColor
         TitleLabel.appearance().textColor = theme.textColor
         TextLabel.appearance().textColor = theme.textColor
+        CircleImageView.appearance().backgroundColor = theme.accentColor
 
         VLInputTextView.appearance().borderColor = theme.inputBorderColor
+
+        // Remove and re-add subviews so that theme is applied to already created UIView instances
```
Comment (connienguyen) [inline VOLA/VOLA/Helpers/Util/ThemeManager.swift]: I tried to do that before but it didn't work, so I used this method that I found here:

https://stackoverflow.com/questions/20875107/force-view-controller-to-reload-to-refresh-uiappearance-changes

File: VOLA/VOLA/Helpers/Extensions/UIKit/UIViewController.swift [hunk 4]
```diff
@@ -107,6 +107,23 @@ extension UIViewController {
     }
 
     /**
+        Show alert that user can dismiss or use as a shortcut to the login flow in order to
+        complete an action that requires a user account (e.g. event registration)
+    */
+    func showLoginAlert() {
```
Comment (connienguyen) [inline VOLA/VOLA/Helpers/Extensions/UIKit/UIViewController.swift]: it doesn't take any arguments though?

Comment (bphenriques) [inline VOLA/VOLA/Helpers/Extensions/UIKit/UIViewController.swift]: I missed the line >.< dunno where it was. Nvm

**Pull request merged**